test/e2e/tests/test_version.py

```diff
@@ -0,0 +1,187 @@
+        replacements["LAMBDA_FILE_NAME"] = LAMBDA_FUNCTION_FILE_ZIP
+        replacements["RESERVED_CONCURRENT_EXECUTIONS"] = "3"
+        replacements["CODE_SIGNING_CONFIG_ARN"] = ""
+        replacements["AWS_REGION"] = get_region()
+
+        # Load function CR
+        resource_data = load_lambda_resource(
+            "function",
+            additional_replacements=replacements,
+        )
+        logging.debug(resource_data)
+
+        # Create k8s resource
+        function_reference = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, "functions",
+            resource_name, namespace="default",
+        )
+
+        # Create lambda function
+        k8s.create_custom_resource(function_reference, resource_data)
+        function_resource = k8s.wait_resource_consumed_by_controller(function_reference)
+
+        assert function_resource is not None
+        assert k8s.get_resource_exists(function_reference)
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        yield (function_reference, function_resource)
+
+        _, deleted = k8s.delete_custom_resource(function_reference)
+        assert deleted
+
+@service_marker
+@pytest.mark.canary
+class TestVersion:
+    def test_smoke(self, lambda_client, lambda_function):
+        (_, function_resource) = lambda_function
+        lambda_function_name = function_resource["spec"]["name"]
+
+        resource_name = random_suffix_name("lambda-version", 24)
+
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["AWS_REGION"] = get_region()
+        replacements["FUNCTION_NAME"] = lambda_function_name
+        replacements["VERSION_NAME"] = resource_name
+        
+        # Load Lambda CR
+        resource_data = load_lambda_resource(
+            "version",
+            additional_replacements=replacements,
+        )
+        logging.debug(resource_data)
+
+        # Create k8s resource
+        ref = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+            resource_name, namespace="default",
+        )
+        k8s.create_custom_resource(ref, resource_data)
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        assert cr is not None
+        assert k8s.get_resource_exists(ref)
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        lambda_validator = LambdaValidator(lambda_client)
+
+        version_number = cr['status']['version']
+
+        # Check version exists
+        assert lambda_validator.version_exists(lambda_function_name, version_number)
+
+        # Delete k8s resource
+        _, deleted = k8s.delete_custom_resource(ref)
+        assert deleted is True
+
+        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+        # Check function version doesn't exist
+        assert not lambda_validator.version_exists(lambda_function_name, version_number)
+
+    def test_version_with_check(self, lambda_client, lambda_function):
```

Comment:
What does check here means?

Reply:
To check that the version we created above got deleted. We call `getFunction` API and pass the Version number along with Function name as we are only deleting a version and not the whole function.